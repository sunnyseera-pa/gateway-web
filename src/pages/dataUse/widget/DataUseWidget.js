import React from 'react';
import { LayoutContent } from '../../../components/Layout';
import Typography from '../../../components/Typography';
import Button from '../../../components/Button';
import Checkbox from '../../../components/Checkbox';

const DataUseWidget = () => {
    return (
        <LayoutContent>
            <div className='accountHeader'>
                <Typography variant='h5'>How to start using your widget</Typography>
                <Typography mb={3}>
                    Below is an interactive example of how your widget could look and be used. Please note that this design is compatible
                    with mobile devices and can be resized to fit anywhere on your website. To start using this widget please accept the
                    terms and conditions by clicking on the 'Get widget' button below.
                </Typography>
                <Button mb={3}>Get widget</Button>
                <Typography>By embedding content in your website or app, you are agreeing to the Terms and Conditions</Typography>
                <Checkbox
                    variant='primary'
                    label='I agree to the HDR Widget Terms and Conditions of use'
                    id='termConditions'
                    mb={4}
                    disabled
                />
                <Typography variant='h6'>Data use widget</Typography>
                <Typography color='grey600'>
                    <i>Click the button below to see your filtered data uses</i>
                </Typography>
            </div>
        </LayoutContent>
    );
};

export default DataUseWidget;
