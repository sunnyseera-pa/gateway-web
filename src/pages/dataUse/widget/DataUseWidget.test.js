import '@testing-library/jest-dom/extend-expect';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import React from 'react';
import DataUseWidget from './DataUseWidget';

let wrapper;

describe('Given the DataUseWidget component', () => {
    describe('When it is rendered', () => {
        beforeAll(() => {
            wrapper = render(<DataUseWidget publisherName='Test Custodian' />, {
                wrapper: Providers,
            });
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        it('Then Title and description should be rendered', async () => {
            expect(screen.queryByText('How to start using your widget')).toBeTruthy();
            expect(
                screen.queryByText(
                    "Below is an interactive example of how your widget could look and be used. Please note that this design is compatible with mobile devices and can be resized to fit anywhere on your website. To start using this widget please accept the terms and conditions by clicking on the 'Get widget' button below."
                )
            ).toBeTruthy();
        });

        it('T&C modal should pop up when `Get Widget` button clicked with `Accept` Button disabled', async () => {
            const getWidgetButton = screen.getByTestId('getWidgetButton');
            expect(getWidgetButton).toBeTruthy();
            fireEvent.click(getWidgetButton);
            await waitFor(() => expect(screen.queryByText('HEALTH DATA RESEARCH UK WIDGET TERMS OF USE')).toBeInTheDocument());
            await waitFor(() => expect(screen.getByTestId('accept-button')).toBeDisabled());
        });
    });
    describe('And when the T&C accepted  is true', () => {
        beforeAll(() => {
            wrapper.rerender(<DataUseWidget publisherName='Test Custodian' accepted />, {
                wrapper: Providers,
            });
        });
        it('Then has DataUseWidget code should be shown and GetWiget button disabled', () => {
            const { rerender, container } = wrapper;
            rerender(<DataUseWidget publisherName='Test Custodian' accepted />);
            expect(wrapper.getByTestId('getWidgetButto')).toBeTruthy();
            // expect(screen.getByTestId('getWidgetButton')).toBeDisabled();
            // expect(
            //     screen.queryByText(
            //         'Copy the source code and share with your development team to display the widget above on your website'
            //     )
            // ).toBeInTheDocument();
        });
    });
});
