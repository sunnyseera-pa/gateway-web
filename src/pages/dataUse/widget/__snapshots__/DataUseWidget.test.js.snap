// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the DataUseWidget component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="ui-LayoutContent row"
  >
    <div
      class="ui-LayoutContent--left col-1"
    />
    <div
      class="ui-LayoutContent--content col-10"
    >
      <div
        class="accountHeader"
      >
        <h5
          class="ui-LayoutBox ui-Typography css-kqk0q3-Typography css-1k6rz1e"
        >
          How to start using your widget
        </h5>
        <p
          class="ui-LayoutBox ui-Typography css-ob39y4-Typography css-2lztm6"
        >
          Below is an interactive example of how your widget could look and be used. Please note that this design is compatible with mobile devices and can be resized to fit anywhere on your website. To start using this widget please accept the terms and conditions by clicking on the 'Get widget' button below.
        </p>
        <button
          class="ui-LayoutBox css-kp49dp-Button css-2lztm6"
          data-testid="getWidgetButton"
          type="button"
        >
          Get widget
        </button>
        <p
          class="ui-LayoutBox ui-Typography css-ob39y4-Typography css-1k6rz1e"
        >
          By embedding content in your website or app, you are agreeing to the Terms and Conditions
        </p>
        <label
          class="ui-Checkbox css-170n6b4 css-27yzq3-Checkbox"
          for="termCo  nditions"
        >
          <input
            disabled=""
            id="termCo  nditions"
            type="checkbox"
          />
          <span
            class="ui-Checkbox__label"
          >
            <span>
              I agree to the HDR Widget Terms and Conditions of use
            </span>
          </span>
        </label>
        <h6
          class="ui-LayoutBox ui-Typography css-18e1tcr-Typography css-1k6rz1e"
        >
          Data Use Widget
        </h6>
        <p
          class="ui-LayoutBox ui-Typography css-1egrwj6-Typography css-1k6rz1e"
        >
          <i>
            Click the button below to see your filtered data uses
          </i>
        </p>
        <hdruk-data-uses
          publisher="Test Custodian"
        />
        <br />
      </div>
    </div>
    <div
      class="ui-LayoutContent--right col-1"
    />
  </div>
</div>
`;
